libc/src/string/memory_utils/utils.h

```diff
@@ -297,7 +297,7 @@ LIBC_INLINE void adjust(ptrdiff_t offset, T1 *__restrict &p1,
                         T2 *__restrict &p2, size_t &count) {
   p1 += offset;
   p2 += offset;
-  count -= offset;
+  count -= static_cast<size_t>(offset);
```

Comment:
`adjust(ptrdiff_t, ...)` has to stay due to: https://github.com/llvm/llvm-project/pull/127523#discussion_r1982078210